planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/src/utilities.cpp

```diff
@@ -673,10 +673,54 @@ bool exists(std::vector<T> vec, T element)
   return std::find(vec.begin(), vec.end(), element) != vec.end();
 }
 
+boost::optional<size_t> findNearestIndexToGoal(
+  const std::vector<autoware_planning_msgs::msg::PathPointWithLaneId> & points,
+  const geometry_msgs::msg::Pose & goal, const int64_t goal_lane_id,
+  const double max_dist = std::numeric_limits<double>::max())
+{
+  if (points.empty()) {
+    return boost::none;
+  }
+
+  size_t min_dist_index;
+  double min_dist = std::numeric_limits<double>::max();
+  {
+    bool found = false;
+    for (size_t i = 0; i < points.size(); ++i) {
+      const double x = points.at(i).point.pose.position.x - goal.position.x;
+      const double y = points.at(i).point.pose.position.y - goal.position.y;
+      const double dist = std::hypot(x, y);
+      if (dist < max_dist && dist < min_dist && exists(points.at(i).lane_ids, goal_lane_id)) {
+        min_dist_index = i;
+        min_dist = dist;
+        found = true;
+      }
+    }
+    if (!found) {
+      return boost::none;
+    }
+  }
+
+  size_t min_dist_out_of_range_index;
+  for (size_t i = min_dist_index;; --i) {
```

Comment:
i!=0
```suggestion
  for (size_t i = min_dist_index; i != 0; --i) {
```

Reply:
i != 0 causes build error. 
(This description has no meaning because i is `size_t`)